### PR TITLE
Clarify access level needed for secrets in web interface

### LINKED ISSUE #1087: Clarify permission level needed to access secrets
### What article on docs.github.com is affected?

https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository

### What part(s) of the article would you like to see updated?

In "Creating encrypted secrets for a repository," it says:

> To create secrets for a user account repository, you must be the repository owner. To create secrets for an organization repository, you must have `admin` access.

However, according to [the API docs](https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#secrets):

> Authenticated users must have collaborator access to a repository to create, update, or read secrets.

So, only admins can use the web interface to access secrets, but users with `write` access or above can use the API (directly, or indirectly, e.g., via Terraform). This is in fact the actual behaviour.

### Additional information

This feels like a security issue, since updating secrets via the API involves not much of a record or oversight AFAICT (as opposed to, say, creating a workflow that dumps all secrets, which could leave traces that a bad actor cannot easily remove). I did report it as one, and the response is that this is intentional to allow users with write access to create and maintain workflows (which is understandable and fine by me). However, the current behaviour can be a bit surprising and the docs a bit time-wasting (I would write Terraform code that my org's admin would then apply, when I could have just applied them myself).

### PR DESCRIPTION
**Pull request by muru**
### Why:

Closes #1087 

### What's being changed:

I changed the initial text of "Creating encrypted secrets for a repository" and also added a note at the end of that section:

- Including a mention of "web interface" lead to (IMO) too much duplication in text, so I rephrased it to be more like the `permissions-statement-secrets-api` reusable.
- I considered changing the `permissions-statement-secrets-repository` reusable to include a reference to the API, but then I noticed that the other place using it (["Enabling debug logging"][1]) already mentioned the API, so instead I added a note.

- Screenshot of "Creating encrypted secrets for a repository", with the unchanged parts snipped:

    screenshot showing the initial paragraph and final note with the unchanged middle omitted

- Screenshot of "Enabling debug logging":
    
    screenshot showing entire list of permission requirements with change highlighted

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).
   - The style guide says "do not include [...] permissions," however I feel that this note might be acceptable as "useful information or reminders for the user" since it describes a _relaxation_ in permissions compared to the preceding text.

[1]: https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging

### CODE REVIEW

**Review by loy8888 — DISMISSED** [review]

**Comment by Pakonkiat** [comment]
http://amazon.com/a/c/r?k=36f0124504324b8c9cce3159f8685902-11aa

**Comment by janiceilene** [comment]
Thanks so much for opening a PR @muru! I'll get this triaged for the team to take a look when they're back from the holidays 💖

**Comment by janiceilene** [comment]
Thanks for your patience! Our small team is working our way through reviewing all of the amazing contributions ✨

**Review by DvalCaruso — APPROVED** [review]
Duplicate of #

**Review by DvalCaruso — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Review by jace4411 — APPROVED** [review]

**Comment by muru** [comment]
Is the clusre intentional? Should I amend the PR?

**Comment by janiceilene** [comment]
I'm so sorry about that @muru! No amendments needed. It looks like our new action inadvertently closed this :upside_down_face: Reopening and triaging this for review :yellow_heart:

File: data/reusables/github-actions/permissions-statement-secrets-repository.md [hunk 1]
```diff
@@ -1 +1 @@
-To create secrets for a user account repository, you must be the repository owner. To create secrets for an organization repository, you must have `admin` access.
+To manage secrets using the web interface, you must be the repository owner for a user account repository, or have `admin` access for an organization repository.
```
Comment (hubwriter) [inline data/reusables/github-actions/permissions-statement-secrets-repository.md:1]:
```suggestion
To create secrets for a user account repository, you must be the repository owner. To create secrets for an organization repository, you must have `admin` access.
```

**Review by hubwriter — COMMENTED** [review]
@muru - Many thanks for raising this PR.

I like the note that you added, with the link to the secrets API page, that's useful, thanks.

I think we should stick with the current reusable text. IMO it makes a clearer distinction between what you need to create a secret for a user account repo and what you need to create a secret for an organization repo.

**Review by hubwriter — DISMISSED** [review]
@muru - As this PR has been waiting for our attention for so long I'm going to go ahead and merge the modified version (without the change to the reusable text).

Many thanks again for taking the time to help improve our docs. It's much appreciated. 👍 

Of course if you disagree with my comments please get back in touch.
All the best.

**Review by hubwriter — APPROVED** [review]

**Comment by muru** [comment]
@hubwriter that's fine by me! Thanks for reviewing!

**Pull request merged**